File: 2026-09/src/day_12/prms_181858/Solution.py
def solution(arr, k):
    seen = set()
    ans = [-1] * k

    idx = 0
    for x in arr:
        if x in seen:
            continue
        ans[idx] = x
        seen.add(x)
        idx += 1

        if idx == k:
            break

    return ans

File: 2026-09/src/day_12/prms_181858/Solution.java
package day_12.prms_181858;

import java.util.*;

class Solution {
    public int[] solution(int[] arr, int k) {
        boolean[] seen = new boolean[1 + 100000];
        int[] ans = new int[k];
        Arrays.fill(ans, -1);

        int idx = 0;
        for (int x : arr) {
            if (seen[x]) continue;
            ans[idx++] = x;
            seen[x] = true;

            if (idx == k) break;
        }

        return ans;
    }
}
